content/5_Plat/Matrix_Expo.mdx

```diff
@@ -124,7 +128,46 @@ int main() {
 	cout << base[0][1];
 }
 ```
+</CPPSection>
+
+<PySection>
+```python
+from typing import List
+
+MOD = 10 ** 9 + 7
+
+def mul(A : List[List[int]], B : List[List[int]], MOD : int) -> List[List[int]]:
+	C = [[0, 0], [0, 0]]
+    	for i in range(2):
+        	for j in range(2):
+			for k in range(2):
+				C[i][j] += ((A[i][k]) * (B[k][j])) % MOD
+    	return C
+
+def power(matrix : List[List[int]], n : int) -> List[List[int]]:
+	if n == 0:
+        	return [[1, 0], [0, 1]]
+    	if n == 1:
+        	return matrix
+    	half = power(matrix, n // 2, MOD)
+    	square = mul(half, half, MOD)
+    	if n % 2 == 0:
+        	return square
+    	return mul(matrix, square, MOD)
```

Comment:
what happened to the indentation here?

Reply:
I dint actually change anything, committed now please check again